### ISSUE TITLE
[FEATURE] Drone Attack Type Indication

### ISSUE THREAD

**Issue by ACatWithaGun** [eternalUnion/UltraPain#12]
# Feature

## This report now only concentrates on this feature
`drones: for the antimatter shots, would be better if they had a tracer before shotting, and personally, i think there should be visual/audio clues, of what it will fire next`

### Additions:

* gabriel (1st encounter): small increase on attack speed and he would throw 3 rotating axes with a spread and 2 spears with a spread as well

* gabriel (2nd encouter): starts with his 2 phase behavior, and increased attack speed on second phase (not sure if first one is even possible)

* sisyphus: attack speed increase, i tested with 1.2 and 1.3 and in both i was able to beat him, only had struggle with 1.3

* (disabled by default) sisyphus attacks becomes unparriable on second phase (except this will hurt and destroy)

* panopticon: does not instanty dies, only after the first boss bar drops, spawns a extra maurice, spawns 4 virtue beams (separetely), 2 sideways, 2 normal beams

### Improvements: 

* v2 (1st encounter): reduced knucleblast range, specially because of the small space in the arena

* cerbeus: (disabled by default) make his attacks unparriable when enraged

**Comment by eternalUnion** [comment]
```
gabriel (1st encounter): small increase on attack speed and he would throw 3 rotating axes with a spread and 2 spears with a spread as well

gabriel (2nd encouter): starts with his 2 phase behavior, and increased attack speed on second phase (not sure if first one is even possible)
```

These 2 are possible, we were planning some features for gabriel but they did not make it into the plugin in time.

```
sisyphus: attack speed increase, i tested with 1.2 and 1.3 and in both i was able to beat him, only had struggle with 1.3

(disabled by default) sisyphus attacks becomes unparriable on second phase (except this will hurt and destroy)
```

Increasing attack speed can be done from config, so I don't think of implementing it (maybe just as a default value). Maybe the nerf can be revoked as well as increased stamina.

```
panopticon: does not instanty dies, only after the first boss bar drops, spawns a extra maurice, spawns 4 virtue beams (separetely), 2 sideways, 2 normal beams
```

Yeah, panopticon should be made into a real fight. Virtue beam may be on all axis as you said.

```
v2 (1st encounter): reduced knucleblast range, specially because of the small space in the arena
```

This is just config again, can only change default values

```
drones: for the antimatter shots, would be better if they had a tracer before shotting, and personally, i think there should be visual/audio clues, of what it will fire next
```

I guess I could implement a blue flash for the beam and yellow flash for the railgun

```
cerbeus: (disabled by default) make his attacks unparriable when enraged
```

I don't agree since it is already too hard to parry when enraged because of the increase in speed which makes the parry period shorter, and it would be deserved if you manage to

**Comment by ACatWithaGun** [comment]
i couldn't really mess a lot with the configs in general, since i was in a rush and didn't had time before i go to work

since i dont know exactly what you mean, im assuming that beam is the maurice explosive beams and railcannon, the sentry antimatter shots

when it's a "beam", i tought of it acting like the mini maurices that flesh prison spawns, and for "railcannon", it makes the beep sound effect and makes that pink flash

making the flash of those colors would confusing and would make people think that it's a parriable/unparriable attack, unless that's the idea

speaking of flesh prison, i forgot to mention some feedback i had over it, i don't know how but i was able to make flesh prison drop to 25% hp because of mini maurices beam attack

**Comment by eternalUnion** [comment]
Parry colors and audio cues you mentioned make more sense, will likely implement them. I did not add anything to flesh prison except the spin insignia attack, I don't know how that happened.

**Comment by ACatWithaGun** [comment]
i'm assuming then, the ultrapain difficulty uses the ID from a higher difficulty, like brutal or UKMD, so the maurices are a feature from the higher difficulties

i think what happened was that flesh prison was doing the cyan bullets attack and a maurice fired at that moment, winch likely deflected the bullets back to it

**Comment by eternalUnion** [comment]
Drone indication part is implemented at https://github.com/eternalUnion/UltraPain/releases/tag/fr12-1.0.2-0

**Comment by ACatWithaGun** [comment]
closing due to implementation, futher suggestions or improvements will be made into a new issue
